e2e/cypress/integration/collapsed_reply_threads/unread_spec.js

```diff
@@ -0,0 +1,87 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+
+// ***************************************************************
+// - [#] indicates a test step (e.g. # Go to a page)
+// - [*] indicates an assertion (e.g. * Check the title)
+// - Use element ID when selecting an element. Create one if none.
+// ***************************************************************
+//
+//  Group: @collapsed_reply_threads
+
+describe('Collapsed Reply Threads', () => {
+    let testTeam;
+    let testUser;
+    let otherUser;
+    let testChannel;
+
+    before(() => {
+        cy.apiUpdateConfig({
+            ServiceSettings: {
+                ThreadAutoFollow: true,
+                CollapsedThreads: 'default_off',
+            },
+        });
+
+        // # Create new channel and other user, and add other user to channel
+        cy.apiInitSetup({loginAfter: true, promoteNewUserAsAdmin: true}).then(({team, channel, user}) => {
+            testTeam = team;
+            testUser = user;
+            testChannel = channel;
+
+            cy.apiSaveCRTPreference(testUser.id, 'on');
+            cy.apiCreateUser({prefix: 'other'}).then(({user: user1}) => {
+                otherUser = user1;
+
+                cy.apiAddUserToTeam(testTeam.id, otherUser.id).then(() => {
+                    cy.apiAddUserToChannel(testChannel.id, otherUser.id);
+                });
+            });
+        });
+    });
+
+    beforeEach(() => {
+        // # Visit channel
+        cy.visit(`/${testTeam.name}/channels/${testChannel.name}`);
+    });
+
+    it('MM-T4144_1 should show a new messages line for an unread thread', () => {
+        // # Post a root post as current user
+        cy.postMessageAs({
+            sender: testUser,
+            message: 'Another interesting post,',
+            channelId: testChannel.id,
+        }).then(({id: rootId}) => {
+            // # Post a reply as other user
+            cy.postMessageAs({
+                sender: otherUser,
+                message: 'Self reply!',
+                channelId: testChannel.id,
+                rootId,
+            });
+
+            // # Click root post
+            cy.get(`#post_${rootId}`).click();
+
+            // * RHS should open and new messages line should be visible
+            cy.get('#rhsContainer').findByTestId('NotificationSeparator').should('exist');
```

Comment:
Got consistent failed test here. Could you please check if all good on you? Thanks!
<img width="546" alt="Screen Shot 2021-08-16 at 9 24 09 PM" src="https://user-images.githubusercontent.com/5334504/129572267-cbeca93b-fc9e-41dc-be81-d53648b29fe6.png">

Reply:
This is still failing on me. I'll ask other SDET to run the test and see what they are getting.

Reply:
@koox00 Finally had a test environment for this. Unfortunately, two of the tests failed. Could you help fix? Thanks!
Test report - https://mm-cypress-report.s3.amazonaws.com/355816443-webapp-pr-8494-webapp-pr-8494-mattermost-enterprise-edition:1a742b1/mochawesome.html
<img width="1163" alt="Screen Shot 2021-08-19 at 8 27 24 PM" src="https://user-images.githubusercontent.com/5334504/130068379-85fc4089-2669-42a3-8abd-d33486d21724.png">

Reply:
will do! thx @saturninoabril

Reply:
@saturninoabril should be fixed now :)

Reply:
Thanks for updating! One last failing test. Please see https://mm-cypress-report.s3.amazonaws.com/356124035-webapp-pr-8494-webapp-pr-8494-mattermost-enterprise-edition:918d9bf/mochawesome.html

![Screen Shot 2021-08-20 at 5 52 56 AM](https://user-images.githubusercontent.com/5334504/130149582-9c576df7-617f-4187-80ba-0f3fba64a17a.png)

Reply:
Hooray! All tests passed now.

https://automation-dashboard.vercel.app/cycles/7e1ad9f5-fb7d-497e-98b8-f7d2984ed110

<img width="930" alt="Screen Shot 2021-08-20 at 6 47 16 PM" src="https://user-images.githubusercontent.com/5334504/130222490-a34c4446-38ab-4bd1-9929-3939d9ea581c.png">